src/app/services/elements-view.service.ts

```diff
@@ -0,0 +1,43 @@
+import { combineLatest, map, Observable } from "rxjs";
+import { inject, Injectable } from "@angular/core";
+import { LoggedUserService } from "../auth/logged-user.service";
+import { DatasetSubscriptionsService } from "../dataset-view/dataset.subscriptions.service";
+
+@Injectable({
+    providedIn: "root",
+})
+export class ElementsViewService {
+    private loggedUserService = inject(LoggedUserService);
+    private datasetSubsService = inject(DatasetSubscriptionsService);
+
+    public viewModeElement(): Observable<ViewModeElement> {
+        return combineLatest([
+            this.loggedUserService.adminPrivelegesChanges,
+            this.datasetSubsService.permissionsChanges,
+        ]).pipe(
+            map(([adminPriveleges, datasetPermissions]) => {
+                if (adminPriveleges) {
+                    return ViewModeElement.ADMIN_MODE;
+                }
+
+                const permissions = datasetPermissions.permissions;
+                if (
+                    permissions.canCommit ||
+                    permissions.canDelete ||
+                    permissions.canRename ||
+                    permissions.canSchedule
+                ) {
+                    return ViewModeElement.PERMISSIONS_MODE;
+                }
+
```

Comment:
These checks will not be the same for every element, we have to somehow declare what the element requires.

Reply:
Fixed